### PR TITLE
Improve the geom_smooth() parameters 

### LINKED ISSUE #6652: Improve the geom_smooth parameters
Smoothing is a complete chapter in the [data scienc](https://rafalab.github.io/dsbook/smoothing.html)e book we are trying to provide tools from in R-Instat.

In ggplot2 geom_smooth is really powerful.  It is already in R-Instat but the geom needs improvements and testing.  This can be done by adding the layer to a line plot.  

Later there is a separate issue on adding it to the main dialogue.  This will be very simple, once the geom is corrected.

The use of the geom is described [here](https://ggplot2.tidyverse.org/reference/geom_smooth.html).  

So:
a) Add` Orientation` to the parameters.  It is either "x" or "y".
b) Correct the occurrence of this problem that occurs when returning to edit the geom settings:
image

c) Correct the addition of the formula settings to the R code.  Currently it adds quotes, and these quotes should be deleted.  They generate an error.
d) Add method.args to the parameters.  This should include some examples, and also allow typing.
e) Add se to the parameters.  It is TRUE or FALSE.  Default is TRUE.
f) the span is an updown from (I think 0 to 1.  I think the default is 0.  Make it 0.3 and allow it to increase or decrease by 0.01, rather than the current 0.1.

### PR DESCRIPTION
**Pull request by HawardKetoyoMsatsi**
Fixes #6652
@rdstern this is ready for review.
@africanmathsinitiative/developers this is ready for review.
I added the `se `parameter however vb decodes if as `ifelse `but the r code generated it is correctly decoded as `se`.

### CODE REVIEW

**Comment by rdstern** [comment]
I think that's right.  So just change the ifelse label to `se` instead.

But I also think there should be a linetype and size parameter in smooth, as there is for the line geom.

**Comment by HawardKetoyoMsatsi** [comment]
@rdstern the above has been fixed

**Review by rdstern — DISMISSED** [review]

File: instat/ucrGeom.vb [hunk 1]
```diff
@@ -1728,24 +1728,28 @@ Public Class ucrGeom
...
         'Can we have  stack, DoDragDrop, fill positions here)
         clsgeom_smooth.AddLayerParameter("method", "list", Chr(34) & "lm" & Chr(34), lstParameterStrings:={Chr(34) & "lm" & Chr(34), Chr(34) & "glm" & Chr(34), Chr(34) & "gam" & Chr(34), Chr(34) & "loess" & Chr(34), Chr(34) & "rlm" & Chr(34)})
         'formula has to be an input and we dont have that currently. its passed in like this formula= y ~ x or  formula= y ~ poly(x, 2) or formula= y ~ log(x) so the user has to type in stuff
-        clsgeom_smooth.AddLayerParameter("formula", "editablelist", Chr(34) & "y ~ x" & Chr(34), lstParameterStrings:={Chr(34) & "y ~ x" & Chr(34), Chr(34) & "y ~ poly(x, 2)" & Chr(34), Chr(34) & "y ~ log(x)" & Chr(34), Chr(34) & " y ~ splines::bs(x,3)" & Chr(34)})
-        clsgeom_smooth.AddLayerParameter("se", "boolean", "TRUE")
+        clsgeom_smooth.AddLayerParameter("formula", "editablelist", "y ~ x", lstParameterStrings:={"y ~ x", "y ~ poly(x, 2)", "y ~ log(x)", " y ~ splines::bs(x,3)"})
+        clsgeom_smooth.AddLayerParameter("se ", "list", "TRUE", lstParameterStrings:={"TRUE", "FALSE"})
```
Comment (lloyddewit) [inline instat/ucrGeom.vb:1738]: Are the spaces required?
```suggestion
        clsgeom_smooth.AddLayerParameter("formula", "editablelist", "y ~ x", lstParameterStrings:={"y ~ x", "y ~ poly(x, 2)", "y ~ log(x)", "y ~ splines::bs(x,3)"})
        clsgeom_smooth.AddLayerParameter("se", "list", "TRUE", lstParameterStrings:={"TRUE", "FALSE"})
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@HawardKetoyoMsatsi thanks, just one very small question

File: instat/ucrGeom.vb (hunk 1 above)
Comment (HawardKetoyoMsatsi) [reply instat/ucrGeom.vb:1738]: Yes, the econd space sfter `se ` is needed. In case the space after `se ` is omitted vb labels `ifelse` instead of `se`

**Comment by HawardKetoyoMsatsi** [comment]
@lloyddewit I have made the required changes please take a look.

File: instat/ucrGeom.vb (hunk 1 above)
Comment (shadrackkibet) [reply instat/ucrGeom.vb]: This is strange. Could you add a comment to alert the next maintainer?

Comment (HawardKetoyoMsatsi) [reply instat/ucrGeom.vb]: @shadrackkibet a comment has been added.

**Review by shadrackkibet — APPROVED** [review]
@rdstern could you re-test and approve if you are happy to merge?

**Review by rdstern — APPROVED** [review]

**Review by dannyparsons — APPROVED** [review]

**Pull request merged**